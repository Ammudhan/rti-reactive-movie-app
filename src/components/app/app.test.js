import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './app';
import {MemoryRouter} from "react-router-dom";
import MovieList from "../movie-list"; //Import the MovieList component for mocking

// Mocking the nested components
jest.mock("../movie-list", () => () => (<movie-list />));

describe('APP Component', () => {

  it('should renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        // TODO - Need to explore more on the Memory Router & Enzyme
        <MemoryRouter initialEntries={['/']}>
          <App />
        </MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should matches snapshot', () => {
    const
        tree = renderer.create(
            <MemoryRouter initialEntries={['/']}>
              <App />
            </MemoryRouter>
        ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});