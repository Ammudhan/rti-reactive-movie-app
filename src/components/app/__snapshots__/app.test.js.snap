// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP Component should matches snapshot 1`] = `
Array [
  <span
    className="row"
  >
    <span
      className="col-12 page-layout d-flex align-items-center justify-content-between"
    >
      <h1
        className="title"
      >
        Pop Movie
      </h1>
      <span
        className="nav-icon"
      />
    </span>
  </span>,
  <div
    className="body-content"
  >
    <movie-list />
  </div>,
  <span
    className="row"
  >
    <span
      className="col-12 page-layout d-flex align-items-center justify-content-center"
    >
      <h1
        className="footer-title"
      >
        Footer
      </h1>
    </span>
  </span>,
]
`;

exports[`APP matches snapshot 1`] = `
Array [
  <span
    className="row"
  >
    <span
      className="col-12 page-layout d-flex align-items-center justify-content-between"
    >
      <h1
        className="title"
      >
        Pop Movie
      </h1>
      <span
        className="nav-icon"
      />
    </span>
  </span>,
  <div
    className="body-content"
  >
    <movie-list />
  </div>,
  <span
    className="row"
  >
    <span
      className="col-12 page-layout d-flex align-items-center justify-content-center"
    >
      <h1
        className="footer-title"
      >
        Footer
      </h1>
    </span>
  </span>,
]
`;
